### PR TITLE
Fix minor code style and doc errors and warnings

### PR DESCRIPTION
**Pull request by heuermh**

### CODE REVIEW

File: utils-metrics/pom.xml [hunk 1]
```diff
@@ -120,6 +120,7 @@
     <dependency>
       <groupId>org.bdgenomics.utils</groupId>
       <artifactId>utils-misc_2.10</artifactId>
+      <scope>compile</scope>
```
Comment (fnothaft) [inline utils-metrics/pom.xml]: Why is this compile scoped?

File: utils-metrics/src/main/scala/org/apache/spark/rdd/InstrumentedOrderedRDDFunctions.scala [hunk 2]
```diff
@@ -28,7 +28,7 @@ import scala.reflect.ClassTag
  */
 class InstrumentedOrderedRDDFunctions[K: Ordering: ClassTag, V: ClassTag](self: RDD[(K, V)]) extends Serializable {
 
-  def sortByKey(ascending: Boolean = true, numPartitions: Int = self.partitions.size): RDD[(K, V)] = self match {
+  def sortByKey(ascending: Boolean = true, numPartitions: Int = self.partitions.length): RDD[(K, V)] = self match {
```
Comment (fnothaft) [inline utils-metrics/src/main/scala/org/apache/spark/rdd/InstrumentedOrderedRDDFunctions.scala]: I'm neutral on this one. I think `size` is defined for all `Iterable`s in Scala, while `length` is not.

File: utils-metrics/pom.xml (hunk 1 above)
Comment (heuermh) [inline utils-metrics/pom.xml]: compile is the default scope; for some reason IDEA was giving me trouble unless I made it explicit.  Probably just user error, I'm an emacs guy

Comment (fnothaft) [inline utils-metrics/pom.xml]: Oh, fascinating. I'm an emacs user as well, so I can't comment on why IDEA was giving you the business.

Comment (fnothaft) [inline utils-metrics/pom.xml]: That being said, if it is just IDEA givin' you the business, I'm inclined to not change it.

File: utils-metrics/src/main/scala/org/apache/spark/rdd/InstrumentedOrderedRDDFunctions.scala (hunk 2 above)
Comment (heuermh) [inline utils-metrics/src/main/scala/org/apache/spark/rdd/InstrumentedOrderedRDDFunctions.scala]: for the scala RDD, ```partitions``` returns an ```Array[Partition]```

Comment (fnothaft) [inline utils-metrics/src/main/scala/org/apache/spark/rdd/InstrumentedOrderedRDDFunctions.scala]: Ah, my bad! Yes, then `length` is probably preferable.

**Comment by fnothaft** [comment]
Merged! Thanks @heuermh!

**Pull request merged**